test/IdentityManager.js

```diff
@@ -392,4 +392,153 @@ contract('IdentityManager', (accounts) => {
       })
     })
   })
+  describe('migration', () => {
+    let newIdenManager
+    beforeEach(done => {
+      identityManager.CreateIdentity(user1, recoveryKey, {from: nobody}).then(tx => {
+        const log = tx.logs[0]
+        assert.equal(log.event, 'IdentityCreated', 'wrong event')
+        proxy = Proxy.at(log.args.identity)
+        return IdentityManager.new(userTimeLock, adminTimeLock, adminRate)
+      }).then((instance) => {
+        newIdenManager = instance
+        //user2 is now a younger owner, while user1 is an olderowner
+        return identityManager.addOwner(proxy.address, user2)
+      }).then(tx => {
+        let log = tx.logs[0]
+        assert.equal(log.event, 'OwnerAdded', 'wrong event')
+        assert.equal(log.args.identity, proxy.address, 'wrong proxy')
+        assert.equal(log.args.owner, user2, 'wrong owner added')
+        assert.equal(log.args.instigator, user1, 'wrong initiator')
+        done()
+      }).catch(e => {
+        console.log(e)
+        done()
+      })
+    })
+
+    it('correct keys can start transfer', (done) => {
+      //olderOwner should be able to start transfer
+      identityManager.initiateMigration(proxy.address, newIdenManager.address, {from: user1}).then(tx => {
+        let log = tx.logs[0]
+        assert.equal(log.event, 'MigrationInitiated', 'wrong event initiated')
+        assert.equal(log.args.identity, proxy.address, 'migrating wrong proxy')
+        assert.equal(log.args.newIdManager, newIdenManager.address, 'migrating to wrong location')
+        assert.equal(log.args.instigator, user1, 'migrating to wrong location')
+        //young owner should not be able to start transfer
+        return identityManager.initiateMigration(proxy.address, newIdenManager.address, {from: user2})
+      }).then(tx => {
+        assert.isFalse(true, 'Should have thrown an error here')
+      }).catch((error) => {
+        assert.isTrue(error.toString().indexOf("invalid JUMP") != -1, 'Should have thrown, did not')
+      }).then(() => {
+        //non-owner should not be able to start transfer
+        return identityManager.initiateMigration(proxy.address, newIdenManager.address, {from: nobody})
+      }).then(() => {
+        assert.isFalse(true, 'Should have thrown an error here')
+      }).catch((error) => {
```

Comment:
When you catch the error like this it will print the error but it's not actually reported as an error (in automated tests etc). I think it should be enough to do `.catch(done)` and the testing framework will take care of printing the error etc.